src/lib/y2network/sysconfig/connection_config_readers/base.rb

```diff
@@ -35,8 +38,41 @@ def initialize(file)
           @file = file
         end
 
+        # Builds a connection configuration object
+        #
+        # @return [Y2Network::ConnectionConfig::Base]
+        def connection_config
+          connection_class.new.tap do |conn|
+            conn.bootproto = BootProtocol.from_name(file.bootproto || "static")
+            conn.description = file.name
+            conn.interface = file.interface
+            conn.ip_configs = ip_configs
+            conn.startmode = Startmode.create(file.startmode || "manual")
+            conn.startmode.priority = file.ifplugd_priority if conn.startmode.name == "ifplugd"
+            update_connection_config(conn)
+          end
+        end
+
       private
 
+        # Returns the class of the connection configuration
+        #
+        # @note This method should be redefined by derived classes.
+        #
+        # @return [Class]
+        def connection_class
+          raise NotImplementedError
```

Comment:
btw you can easily compute it from self.class. So you can have something like 
```
class_name = self.class.to_s[/::([^:])*$/, 1]
Y2Network::ConnectionConfig.const_get(class_name)
```

this avoids to redefine it everywhere.